articles/directory-structure.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Directory Structure
+layout: page
+---
+
+## Directory Structure <!-- omit in toc -->
+
+- [Base directory](#base-directory)
+  - [/config](#config)
+  - [/src](#src)
+  - [/static](#static)
+  - [/test](#test)
+- [The /src directory](#the-src-directory)
+  - [/app](#app)
+  - [/assets](#assets)
+  - [/exceptions](#exceptions)
+  - [/messages](#messages)
+  - [/providers](#providers)
+
+### Base directory
+
+#### `/config`
+
+A folder containing a single `index.ts` file that exports the application
+config. Read more about configuration in the
+[Application Configuration][application-configuration] article.
+
+#### `/src`
+
+This folder contains all your extension's source code. All of its contents and
+subfolders will be covered in more detail in the next section of this article.
+
+#### `/static`
+
+A space dedicated to store your static assets, like images or documents. These
+assets can then be accessed from the codebase via the [`Extension`][Extension]
+APIs.
+
+#### `/test`
+
+The `test` directory contains a unit test scaffold based on [`mocha`][mocha].
+
+---
+
+### The `/src` directory
+
+#### `/app`
+
+The `app` folder contains all main application logic that **needs to be split**
```

Comment:
What about "needs to be separated"?

Reply:
eh doesn't really go well with the end of the sentence